persistence/persistence-mybatis-test/src/test/java/org/cbioportal/persistence/mybatis/ClinicalDataMyBatisRepositoryTest.java

```diff
@@ -0,0 +1,508 @@
+        BaseMeta result = clinicalDataMyBatisRepository.getMetaSampleClinicalData(studyIds, sampleIds,
+                "OTHER_SAMPLE_ID");
+
+        Assert.assertEquals((Integer) 1, result.getTotalCount());
+    }
+
+    @Test
+    public void getMetaSampleClinicalDataMultipleStudyMultipleSampleNullAttribute() throws Exception {
+
+        List<String> studyIds = new ArrayList<>();
+        studyIds.add("study_tcga_pub");
+        studyIds.add("acc_tcga");
+        List<String> sampleIds = new ArrayList<>();
+        sampleIds.add("TCGA-A1-A0SB-01");
+        sampleIds.add("TCGA-A1-B0SO-01");
+        BaseMeta result = clinicalDataMyBatisRepository.getMetaSampleClinicalData(studyIds, sampleIds, null);
+
+        Assert.assertEquals((Integer) 5, result.getTotalCount());
+    }
+
+    @Test
+    public void getAllClinicalDataOfPatientInStudySingleStudyNullPatientNullAttributeIdProjection() throws Exception {
+
+        List<String> studyIds = new ArrayList<>();
+        studyIds.add("study_tcga_pub");
+        List<PatientClinicalData> result =
+                clinicalDataMyBatisRepository.getAllClinicalDataOfPatientInStudy(studyIds, null, null, "ID", null, null,
+                        null, null);
+
+        Assert.assertEquals(4, result.size());
+        PatientClinicalData data = result.get(0);
+        Assert.assertEquals("FORM_COMPLETION_DATE", data.getAttrId());
+        Assert.assertEquals((Integer) 1, data.getInternalId());
+        Assert.assertNull(data.getClinicalAttribute());
+        Assert.assertNull(data.getPatient());
+        Assert.assertNull(data.getAttrValue());
+    }
+
+    @Test
+    public void getAllClinicalDataOfPatientInStudySingleStudyNullPatientNullAttributeSummaryProjection()
+            throws Exception {
+
+        List<String> studyIds = new ArrayList<>();
+        studyIds.add("study_tcga_pub");
+        List<PatientClinicalData> result =
+                clinicalDataMyBatisRepository.getAllClinicalDataOfPatientInStudy(studyIds, null, null, "SUMMARY", null,
+                        null, null, null);
+
+        Assert.assertEquals(4, result.size());
+        PatientClinicalData data = result.get(0);
+        Assert.assertEquals("FORM_COMPLETION_DATE", data.getAttrId());
+        Assert.assertEquals("2013-12-5", data.getAttrValue());
+        Assert.assertEquals((Integer) 1, data.getInternalId());
+        Assert.assertNull(data.getClinicalAttribute());
+        Assert.assertNull(data.getPatient());
+    }
+
+    @Test
+    public void getAllClinicalDataOfPatientInStudySingleStudyNullPatientNullAttributeDetailedProjection()
+            throws Exception {
+
+        List<String> studyIds = new ArrayList<>();
+        studyIds.add("study_tcga_pub");
+        List<PatientClinicalData> result =
+                clinicalDataMyBatisRepository.getAllClinicalDataOfPatientInStudy(studyIds, null, null, "DETAILED", null,
+                        null, null, null);
+
+        Assert.assertEquals(4, result.size());
```

Comment:
Since we are testing the correctness of a SQL select query, I think it's enough to check the size and first element, because you can't really return a different kind of second or third element from a select query.